docs/4.x/tracking-optout.md

```diff
@@ -0,0 +1,156 @@
+---
+category: Integrate
+title: Tracking Opt-out
+--- 
+# Opt-Out
+
+If you want to provide your visitors with the choice to opt-out of Matomo tracking rather than [asking for consent](/guides/tracking-consent) then Matomo
+includes a built-in opt-out form that you can add to your website. 
+
+This involves adding small amount of HTML/JavaScript to the appropriate pages on your site. This code can be generated from the Administration
+=> Privacy => Users opt-out menu in the Matomo dashboard. The generated code can include custom CSS styles and optionally skip the introduction text.
+
+The opt-out form code can be generated to work in two ways, either using the Matomo tracker code or as self-contained code, each option has advantages and 
+disadvantages.
+
+## Opt-out using the Matomo tracker code
+
+Using this option the embedded opt-out form code will load Javascript from your Matomo instance, this JavaScript will then create the opt-out form in the designated 
+`<div>` container. If the visitor opts in or out then the Matomo tracking code functions will be preferred to set the consent choice.
+
+### Process
+
+- The embedded website `<script>` tag requests  the `optOutJS` JavaScript from the Matomo instance, passing admin configuration options as URL parameters. 
+- The Matomo instance translates all form text to into the chosen language, or the browser language if `language=auto` is used and returns the resulting JavaScript.
+- The opt-out JavaScript executes when the `DOMContentLoaded` event is fired, indicating the page has finished loading.
+- It checks that the specified opt-out form div exists on the page, if not then an error is shown.
+- It then waits for the Matomo Tracker code to become available.
+  - If the Matomo tracker code is found then the opt-out form is created to use the Matomo tracker code to set consent.
+  - If the Matomo Tracker code does not become available with the timeout period then the opt-out form will be created to set consent cookies directly.
+- If cookies are disabled in the browser or the connection is not `HTTPS` then an error will be shown. 
```

Comment:
a console error?

Reply:
The no cookies / no https error is shown on the page in the content div. Even if the content div is missing one will be added, the idea is to avoid failing silently with no opt-out shown.